doc/Makefile.am

```diff
@@ -99,14 +99,44 @@ doxygen_sources = \
 	$(top_srcdir)/tagmanager/src/tm_source_file.[ch] \
 	$(top_srcdir)/tagmanager/src/tm_workspace.[ch]
 
-Doxyfile.stamp: Doxyfile $(doxygen_sources)
-	$(AM_V_GEN)$(DOXYGEN) Doxyfile && echo "" > $@
+# set WARN_IF_UNDOCUMENTED because apparently doxygens warns for undocumented stuff
+# in headers (even though it's correctly documented in the corresponding .c file) only
+# for xml output
+Doxyfile-gi: Doxyfile
+	$(AM_V_GEN)$(SED) \
+		-e 's,addtogir=@internal,addtogir=,' \
+		-e 's,^\(GENERATE_HTML.*\)YES,\1NO,' \
+		-e 's,^\(GENERATE_XML.*\)NO,\1YES,' \
+		-e 's,^\(WARN_IF_UNDOCUMENTED.*\)YES,\1NO,' \
+		-e 's,^\(SORT_MEMBER_DOCS.*\)YES,\1NO,' \
+		-e 's,^\(SORT_BRIEF_DOCS.*\)YES,\1NO,' \
+		$< > $@ || { $(RM) $@ && exit 1; }
 
-all-local: Doxyfile.stamp
+Doxyfile.stamp: Doxyfile Doxyfile-gi $(doxygen_sources)
+	$(AM_V_GEN)$(DOXYGEN) Doxyfile-gi && $(DOXYGEN) Doxyfile && echo "" > $@
+
+ALL_TARGETS = Doxyfile.stamp
```

Comment:
why introducing this new variable?  AFAIK it's documented Make behavior to *add* dependencies when specifying a rule with only dependencies and no rules, so it would do the same as you do here… or am I missing some Automake weirdness?